Add "Custom sort in File Explorer" plugin

SebastianMC opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/SebastianMC/obsidian-custom-sort

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [x]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE.md file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
["authorUrl"](https://github.com/SebastianMC/obsidian-custom-sort/blob/57b1cd8ae2404c71e261826512c2cb1d3f67f7ad/manifest.json#L8)
Are you sure that you want to expose your email address like that?
It will be visible when browsing community plugins.

[const fileExplorer = this.app.viewRegistry.viewByType["file-explorer"](leaf) as FileExplorerView;](https://github.com/SebastianMC/obsidian-custom-sort/blob/57b1cd8ae2404c71e261826512c2cb1d3f67f7ad/src/main.ts#L239)
You should use the `getFileExplorer()` function you created below.

[_l1s5_processParsedSortGroupSpec](https://github.com/SebastianMC/obsidian-custom-sort/blob/57b1cd8ae2404c71e261826512c2cb1d3f67f7ad/src/custom-sort/sorting-spec-processor.ts#L695)
What is the reasoning for these function names?
They are not really descriptive.

SebastianMC commented:
Thank you @joethei for the valuable remarks!

Code of the plugin was updated and released, the updates include:
- removed email from the source code. Indeed I'm new to public projects and repositories and not (yet) so sensitive to the privacy aspects. Most of my prior projects were not public
- unified the way of accessing FileExplorerView: all of the access goes via `getFileExplorer()`
- removed the confusing prefixes like `_l1s3_....` from parser methods. They were a residue from early development stage naming. Originally it allowed clear ordering of these functions in the code. No longer needed

liamcain commented:
- [additionalSortspecFile: 'Inbox/Inbox.md',](https://github.com/SebastianMC/obsidian-custom-sort/blob/d672df6303b221d1907526b3d15171560b89ede2/src/main.ts#L37-L38) You might want to make the default value for this `''`. Then check if the setting is configured before reading anything. Inbox.md is pretty generic so it might be unexpected for this plugin to read it by default.
